### PR TITLE
Spreadsheet: Add mathematical and logical functions

### PR DESCRIPTION
**Pull request by ryangjchandler**
This pull request introduces a handful of useful mathematical and logical functions.

* `abs` - the absolute value of a number.
* `isEven` - checks whether the value is even.
* `isOdd` - checks whether the value is odd.
* `ln(x)` - calculates the natural log of a number (log base `e` of `x`)
* `log(x, base)` - calculates the log of `x` to base `base`.

### CODE REVIEW

File: Base/res/js/Spreadsheet/runtime.js [hunk 1]
```diff
@@ -474,6 +474,21 @@ function reflookup(req_lookup_value, lookup_inputs, lookup_outputs, if_missing,
     );
 }
 
+function abs(value) {
+    return Math.abs(value)
+}
+
+abs.__documentation = JSON.stringify({
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:497]: Please put this after the comment right below it.

File: Base/res/js/Spreadsheet/runtime.js [hunk 2]
```diff
@@ -474,6 +474,21 @@ function reflookup(req_lookup_value, lookup_inputs, lookup_outputs, if_missing,
     );
 }
 
+function abs(value) {
+    return Math.abs(value)
+}
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:479]: This feels kinda pointless, why yoink this out of the Math namespace?
Generally free functions in here should make sense on a range, not a single value, so I don't really think this function is even needed (as the user can just as easily write `Math.abs`)

File: Base/res/js/Spreadsheet/runtime.js [hunk 3]
```diff
@@ -478,6 +478,14 @@ function abs(value) {
     return Math.abs(value)
 }
 
+function isEven(value) {
+    return value % 2 === 0
+}
+
+function isOdd(value) {
+    return ! isEven(value)
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:486]: Please run prettier on this file.

**Comment by ryangjchandler** [comment]
Happy to remove the less useful ones.

**Review by alimpfard — COMMENTED** [review]
While these functions are _fine_ to have, they should at least properly support operating on ranges.

**Pull request closed without merging**